azkaban-common/src/main/java/azkaban/executor/container/ContainerImplUtils.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2021 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.executor.container;
+
+import azkaban.executor.ExecutableFlow;
+import azkaban.executor.ExecutableFlowBase;
+import azkaban.executor.ExecutableNode;
+import azkaban.executor.ExecutorManagerException;
+import java.util.Collections;
+import java.util.Set;
+import java.util.TreeSet;
+import java.util.stream.Collectors;
+
+public class ContainerImplUtils {
+
+  private ContainerImplUtils() {
+    // Not to be instantiated
+  }
+
+  /**
+   * This method is used to get jobTypes for a flow. This method is going to call
+   * populateJobTypeForFlow which has recursive method call to traverse the DAG for a flow.
+   *
+   * @param flow Executable flow object
+   * @return
+   * @throws ExecutorManagerException
+   */
+  public static TreeSet<String> getJobTypesForFlow(final ExecutableFlow flow) {
+    final TreeSet<String> jobTypes = new TreeSet<>();
+    populateJobTypeForFlow(flow, jobTypes);
+    return jobTypes;
+  }
+
+  /**
+   * This method is used to populate jobTypes for ExecutableNode.
+   *
+   * @param node
+   * @param jobTypes
+   */
+  public static void populateJobTypeForFlow(final ExecutableNode node, Set<String> jobTypes) {
+    if (node instanceof ExecutableFlowBase) {
+      final ExecutableFlowBase base = (ExecutableFlowBase) node;
+      for (ExecutableNode subNode : base.getExecutableNodes()) {
+        populateJobTypeForFlow(subNode, jobTypes);
+      }
+    } else {
+      jobTypes.add(node.getType());
+    }
+  }
+
+  public static Set<String> getAllowedJobTypes(Set<String> allowList, Set<String> denyList) {
+    if (null == allowList && null == denyList) {
+      return Collections.emptySet();
+    }
+    if (null == allowList) {
```

Comment:
Why will denyList be returned if allowList is not present? It should be emptySet right?

Reply:
Thanks. Corrected it.